### PR TITLE
[sk] Added templates for transformer blocks

### PR DESCRIPTION
**Pull request by skunichetty**
# Summary
Added templates for transformer blocks:
- Default template: empty template, allows user to define whatever execution logic they like
- Transformer action template: transformer block that uses the `transformer-actions` library
- Template generation from cleaning suggestion: given a cleaning suggestion produced by the `data_cleaner` library, `build_template_from_suggestion` generates the corresponding template that can be written to file. This can be used once the ability to convert cleaning suggestions to blocks is finished.

Alongside these templates, a couple of refactoring changes were made to the `transformer_actions` library
- Previous collections of constants were converted to enumerations (with string mixins to ensure backwards compatibility) to provide namespaced access to constants. This makes accessing the possible values for parameters like `ActionType` and `Axis` easier for users.
- The `_build_action_variables` instance method from `cleaning_rules.BaseRule` was refactored into a standalone method in `transformer_actions.utils` so that it can be used by templates for generating the action variable information for a transformer action payload.
- A new `build_transformer_action` method was created in `transformer_actions.utils` to automatically build a payload from parameters and the input data frame. This method abstracts out action variable information so users don't have to manually generate that information.

# Tests
Previous unit tests were ran to confirm that none of the `data_cleaner` library's functionality was broken in this latest update. New unit tests were written to test that template generation for cleaning suggestions is correct.

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by skunichetty** [comment]
Hold off on reviewing this for now, I'm going to make a few updates with regards to how some of the template generation features are handled.

**Comment by skunichetty** [comment]
Nevermind on that update actually, I'll just make a new PR with the updated code.

File: mage_ai/data_cleaner/transformer_actions/utils.py [hunk 1]
```diff
@@ -1,15 +1,92 @@
-from mage_ai.data_cleaner.column_types.column_type_detector import REGEX_NUMBER
+from pandas import DataFrame
```
Comment (wangxiaoyou1993) [inline mage_ai/data_cleaner/transformer_actions/utils.py:1]: nit: alphabetical order

File: mage_ai/data_preparation/templates/transformers/transformer_action.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    Template code for transforming data frame using actions from
+    the `transformer_actions` library.
+
+    See library documentation for the required payload parameters when constructing
+    the action below.
+    """
+    action = build_transformer_action(
+        action_type=ActionType.CUSTOM,  # Specify the transformer action to perform
+        action_arguments=[],  # Specify the columns of data frame to apply the action tos
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/templates/transformers/transformer_action.py:18]: nit: remove the 's' at the end

File: mage_ai/data_preparation/templates/utils.py [hunk 3]
```diff
@@ -10,3 +12,33 @@ def copy_templates(template_path: str, dest_path: str) -> None:
...
+    Args:
+        suggestion (Dict): Suggestion payload generated from `BaseRule`.
+
+    Returns:
+        str: String version of Python code to execute to run this cleaning suggestion.
+    """
+    clean_title = suggestion['title'].lower().replace(' ', '_')
+    cleaned_payload = json.dumps(suggestion['action_payload'], indent=4)
+    cleaned_payload = cleaned_payload.replace('\n', '\n    ')
+    template = f"""from mage_ai.data_cleaner.transformer_actions.base import BaseAction
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/templates/utils.py:30]: You can put the template in a file and then render the file with the suggestion input.
Maybe take a look at the jinja template

Comment (skunichetty) [reply mage_ai/data_preparation/templates/utils.py:30]: I was exactly thinking this method and I'm currently working on a similar solution, but I didn't think to use Jinja. I'll look into using jinja then.

Comment (skunichetty) [reply mage_ai/data_preparation/templates/utils.py]: Switched to Jinja for loading template

File: mage_ai/data_preparation/templates/transformers/transformer_action.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from pandas import DataFrame
+
+
+@transformer
+def execute_transformer_action(df: DataFrame) -> DataFrame:
+    """
+    Template code for transforming data frame using actions from
+    the `transformer_actions` library.
+
+    See library documentation for the required payload parameters when constructing
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/templates/transformers/transformer_action.py:13]: nit: add link or path to the library documentation

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**